src/esp/assets/ResourceManager.cpp

```diff
@@ -1646,33 +1746,50 @@ void ResourceManager::addPrimitiveToDrawables(int primitiveID,
-                        meshAttributeFlags,  // meshAttributeFlags
-                        node,                // scene node
-                        NO_LIGHT_KEY,        // lightSetup key
-                        WHITE_MATERIAL_KEY,  // material key
-                        drawables);          // drawable group
+  createDrawable(*primitive_meshes_.at(primitiveID),  // render mesh
+                 meshAttributeFlags,                  // meshAttributeFlags
+                 node,                                // scene node
+                 NO_LIGHT_KEY,                        // lightSetup key
+                 WHITE_MATERIAL_KEY,                  // material key
+                 drawables);                          // drawable group
 }
 
 void ResourceManager::removePrimitiveMesh(int primitiveID) {
   CHECK(primitive_meshes_.count(primitiveID));
   primitive_meshes_.erase(primitiveID);
 }
 
-void ResourceManager::createGenericDrawable(
-    Mn::GL::Mesh& mesh,
-    gfx::Drawable::Flags& meshAttributeFlags,
-    scene::SceneNode& node,
-    const Mn::ResourceKey& lightSetupKey,
-    const Mn::ResourceKey& materialKey,
-    DrawableGroup* group /* = nullptr */) {
-  node.addFeature<gfx::GenericDrawable>(
-      mesh,                // render mesh
-      meshAttributeFlags,  // mesh attribute flags
-      shaderManager_,      // shader manager
-      lightSetupKey,       // kightSetup key
-      materialKey,         // material key
-      group);              // drawable group
+void ResourceManager::createDrawable(Mn::GL::Mesh& mesh,
+                                     gfx::Drawable::Flags& meshAttributeFlags,
+                                     scene::SceneNode& node,
+                                     const Mn::ResourceKey& lightSetupKey,
+                                     const Mn::ResourceKey& materialKey,
+                                     DrawableGroup* group /* = nullptr */) {
+  const auto& materialDataType =
+      shaderManager_.get<gfx::MaterialData>(materialKey)->type;
+  switch (materialDataType) {
+    case gfx::MaterialDataType::Phong:
+      node.addFeature<gfx::GenericDrawable>(
+          mesh,                // render mesh
+          meshAttributeFlags,  // mesh attribute flags
+          shaderManager_,      // shader manager
+          lightSetupKey,       // lightSetup key
+          materialKey,         // material key
+          group);              // drawable group
+      break;
+    case gfx::MaterialDataType::Pbr:
+      node.addFeature<gfx::PbrDrawable>(
+          mesh,                // render mesh
+          meshAttributeFlags,  // mesh attribute flags
+          shaderManager_,      // shader manager
+          lightSetupKey,       // lightSetup key
+          materialKey,         // material key
+          group);              // drawable group
+      break;
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
```

Comment:
Continuing from [the disccussion here](https://github.com/facebookresearch/habitat-sim/pull/812#discussion_r499002662) -- I want to understand your intention better:

> I just would like to get rid of such unnecessary warning so I used "default".

Why do you think it's unnecessary? If somebody adds a new material data type to the enum later (`Flat` or `PbrSpecularGlossiness` let's say), it will helpfully tell them what `switch`es need an update. What's unnecessary on that? 

And if the enum contains other values you don't want to handle, you'd just explicitly list them:

```cpp
  switch (materialDataType) {
    case gfx::MaterialDataType::Phong:
      node.addFeature(...);
      return;
    case gfx::MaterialDataType::Pbr:
      node.addFeature(...);
      return;

    case gfx::MaterialDataType::None:
      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
}

CORRADE_INTERNAL_ASSERT_UNREACHABLE();
```

Reply:
changed.